ground/src/main/java/com/google/android/ground/repository/UserRepository.kt

```diff
@@ -54,7 +55,9 @@ constructor(
       Timber.d("Skipped refreshing user profile as device is offline.")
       return
     }
-    remoteDataStore.refreshUserProfile()
+    if (!USE_EMULATORS) {
+      remoteDataStore.refreshUserProfile()
+    }
```

Comment:
Instead of exposing USE_EMULATORS to the repositories, can we instead create a custom annotation (something like `@IsAnonymousSession`), provide it in the authentication module and use it here and everywhere else?

Reply:
How would the annotation be used? Is there a reason not to reference the global BuildConfig in repositories? Testability?

Reply:
The main intention behind the suggestion is to enhance readability. Right now some logic around what to do is hidden behind this boolean. The repositories shouldn't care whether the app is running in an emulator or not. All it cares about is whether the session is anonymous or not. 

Then we can add some test coverage around it easily as well.

Reply:
Still not clear how the annotation helps, since they are static bindings, while user sessions are transient. In this case, perhaps just checking whether the current user is anonymous here is sufficient?